sphinxcontrib/dotnetdomain.py

```diff
@@ -571,6 +573,23 @@ class DotNetDomain(Domain):
         DotNetIndex,
     ]
 
+    def __init__(self, *args, **kwargs):
+        super(DotNetDomain, self).__init__(*args, **kwargs)
+        self._role2type = {}
+        self._type2role = {}
+        for name, obj in iteritems(self.object_types):
+            for roles in obj.roles:
+                if isinstance(roles, string_types):
+                    roles = [roles]
+                try:
+                    for rolename in roles:
+                        self._role2type.setdefault(rolename, []).append(name)
+                except TypeError:
+                    pass
+            self._type2role[name] = obj.roles[0] if obj.roles else ''
```

Comment:
Will this not be an issue when obj.roles is a `string_type`, which the check above is looking for?